src/index.js

```diff
@@ -74,18 +83,40 @@ class ProfilerPlugin {
   }
 
   async postInvoke() {
-    try {
-      if (!getEnabledStatus(this.enabled)) return;
+    if (!(this.profilerEnabled || this.heapsnapshotEnabled)) return false;
+
+    return new Promise(async resolve => {
+      try {
+        const signedRequestURL = await this.getSignedUrl();
+        const archive = archiver.default('zip');
+        const archiveBuffer = [];
+
+        archive.on('data', chunk => {
+          archiveBuffer.push(chunk);
+        });
+        archive.on('finish', async () => {
```

Comment:
Note why we cannot use streams here (cannot do PUT into S3 signed urls with chucked encoding as a Stream would enforce)

Reply:
done